internal/controller/nopresource/nopresource.go

```diff
@@ -128,13 +97,70 @@ type external struct {
 }
 
 func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mg.(*v1alpha1.MyType)
+	cr, ok := mg.(*v1alpha1.NopResource)
 	if !ok {
-		return managed.ExternalObservation{}, errors.New(errNotMyType)
+		return managed.ExternalObservation{}, errors.New(errNotNopResource)
+	}
+	startTime := cr.CreationTimestamp
+
+	// If object was deleted, return it does not exist so that managed reconciler removes finalizer
+	if meta.WasDeleted(mg) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	// Logic 1 - Set condition as soon as the time elapsed in seconds is equal to time specified in Spec
```

Comment:
@hasheddan this logic will set the condition as soon as the time elapsed in seconds is equal to time specified in Spec
The behavior of the same can be found at `logs/logic1. log`